rotkehlchen/chain/ethereum/modules/yearn/decoder.py

```diff
@@ -0,0 +1,128 @@
+            self,
+            token: 'EvmToken',
+            tx_log: EvmTxReceiptLog,  # pylint: disable=unused-argument
+            transaction: EvmTransaction,
+            event: HistoryBaseEntry,
+            action_items: list[ActionItem],  # pylint: disable=unused-argument
+    ) -> bool:
+        counterparty = CPT_YEARN_V2
+        if transaction.to_address in self.vaults_v1:
+            counterparty = CPT_YEARN_V1
+        elif transaction.to_address not in self.vaults_v2:
+            return False
+
+        is_deposit = False
+        if transaction.input_data.startswith(YEARN_DEPOSIT_4_BYTES):
+            is_deposit = True
+        elif not transaction.input_data.startswith(YEARN_WITHDRAW_4_BYTES):
+            # in this case it is a method that we don't handle
+            return False
+
+        if (
+            is_deposit is True and
+            event.event_type == HistoryEventType.SPEND and
+            event.counterparty == transaction.to_address
+        ):
+            event.event_type = HistoryEventType.DEPOSIT
+            event.event_subtype = HistoryEventSubType.DEPOSIT_ASSET
+            event.counterparty = counterparty
+            vault_token_name = _get_vault_token_name(transaction.to_address)
+            event.notes = f'Deposit {event.balance.amount} {token.symbol} in {counterparty} vault {vault_token_name}'  # noqa: E501
+        elif (
+            is_deposit is True and
+            event.event_type == HistoryEventType.RECEIVE and
+            event.counterparty == ZERO_ADDRESS
+        ):
+            event.event_type = HistoryEventType.DEPOSIT
+            event.event_subtype = HistoryEventSubType.RECEIVE_WRAPPED
+            event.counterparty = counterparty
+            vault_token_name = _get_vault_token_name(transaction.to_address)
+            event.notes = f'Receive {event.balance.amount} {vault_token_name} after deposit in a {counterparty} vault'  # noqa: E501
+        elif (
+            is_deposit is False and
+            event.event_type == HistoryEventType.RECEIVE and
+            event.counterparty == transaction.to_address
+        ):
+            event.event_type = HistoryEventType.WITHDRAWAL
+            event.event_subtype = HistoryEventSubType.REMOVE_ASSET
+            event.counterparty = counterparty
+            vault_token_name = _get_vault_token_name(transaction.to_address)
+            event.notes = f'Withdraw {event.balance.amount} {token.symbol} from {counterparty} vault {vault_token_name}'  # noqa: E501
+        elif (
+            is_deposit is False and
+            event.event_type == HistoryEventType.SPEND and
+            event.counterparty == ZERO_ADDRESS
+        ):
+            event.event_type = HistoryEventType.WITHDRAWAL
+            event.event_subtype = HistoryEventSubType.RETURN_WRAPPED
+            event.counterparty = counterparty
+            event.notes = f'Return {event.balance.amount} {token.symbol} to the {counterparty} vault'  # noqa: E501
```

Comment:
Also can't you get the vault name here?

```suggestion
            event.notes = f'Return {event.balance.amount} {token.symbol} to a {counterparty} vault'  # noqa: E501
```